### DISCUSSION TITLE
Wait for Modal close in chained actions

### DISCUSSION THREAD

**Discussion by sql4bucks** [Budibase/budibase#8601] [Help] [answered]
I'm trying to:

1. Open a modal from the actions defined on a button in the parent screen
2. Wait for the user to select and set state data in the modal 
3. Then proceed with additional actions after the modal is closed. 

I defined all of these chained actions in the button on the parent screen. 

My problem is that the chained actions AFTER opening the modal don't wait for the modal  close before executing; they execute asynchronously while the modal is still open. I need to wait for the user to select data on the modal before proceeding to the next actions. I thought the nature of a modal window would include "block and wait" behavior.

If this is not possible, what is the "Budibase way" of executing this type of logic?

**Comment by melohagan** [comment, marked as answer]
Hey @sql4bucks 

This would be expected behaviour as the action is simply to open the modal, then continue onto the subsequent actions.

What you may want to do is have the last action in your parent screen be open screen as modal. 
Then in your modal screen you could set the state in the modal via a button click, and this state should proxy back and be accessible in the parent screen.

**Reply by sql4bucks** [reply]
@melohagan - Thanks for the quick reply
